packages/ai-jsx/src/core/errors.ts

```diff
@@ -0,0 +1,10 @@
+export class HttpError extends Error {
+  constructor(
+    message: string,
+    readonly statusCode: number,
+    readonly responseBody?: string,
+    readonly responseHeaders?: Record<string, string>
```

Comment:
Yeah I was originally going to include them but the footgun I was trying to avoid was unconditionally logging api keys at the error level.

I think the escape hatch is that components can (do, in the case of OpenAI components) log these details at more verbose levels.